src/java/main/org/apache/zookeeper/server/quorum/QuorumCnxManager.java

```diff
@@ -462,7 +464,7 @@ public void receiveConnection(final Socket sock) {
         DataInputStream din = null;
         try {
             din = new DataInputStream(
-                    new BufferedInputStream(sock.getInputStream()));
+                    new BufferedInputStream(new SocketInputStream(sock, self.tickTime * self.initLimit)));
```

Comment:
All of these timeouts based on initLimit seem possibly wrong. I guess it's better to be consistent for now, but it feels like you want the socket timeouts to be less than the actual "operation" timeout.

Reply:
I've used the same value as for `setSoTimeout` calls, while I agree that it can be smaller it looks like a bigger change.